File: agent/src/test/java/com/teamscale/jacoco/agent/DummyLogger.java
package com.teamscale.jacoco.agent;

import com.teamscale.report.util.ILogger;

/** Implements {@link ILogger} as NOP actions. */
public class DummyLogger implements ILogger {
	@Override
	public void debug(String message) {
		// nothing to do here
	}

	@Override
	public void info(String message) {
		// nothing to do here
	}

	@Override
	public void warn(String message) {
		// nothing to do here
	}

	@Override
	public void warn(String message, Throwable throwable) {
		// nothing to do here
	}

	@Override
	public void error(Throwable throwable) {
		// nothing to do here
	}

	@Override
	public void error(String message, Throwable throwable) {
		// nothing to do here
	}
}

File: agent/src/test/kotlin/com/teamscale/jacoco/agent/DummyLogger.kt
package com.teamscale.jacoco.agent

import com.teamscale.report.util.ILogger

/** Implements [ILogger] as NOP actions.  */
class DummyLogger : ILogger {
    override fun debug(message: String) {
        // nothing to do here
    }

    override fun info(message: String) {
        // nothing to do here
    }

    override fun warn(message: String) {
        // nothing to do here
    }

    override fun warn(message: String, throwable: Throwable) {
        // nothing to do here
    }

    override fun error(throwable: Throwable) {
        // nothing to do here
    }

    override fun error(message: String, throwable: Throwable) {
        // nothing to do here
    }
}
